console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,93 @@
+package v1
+
+import (
+	authorizationv1 "openshift/api/pgk/authorization/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleQuickStart is an extension for guiding user thought various
+// workflows in the OpenShift web console.
+type ConsoleQuickStart struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ConsoleQuickStartSpec `json:"spec"`
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	DisplayName string `json:"displayName"`
```

Comment:
For required fields, I think we need a validation regex so the user can't pass the empty string `""`. IIRC, the empty string passes validation even if required.